File: library/src/main/java/org/fbme/lib/st/expressions/BinaryOperation.java
package org.fbme.lib.st.expressions;

public enum BinaryOperation {
    ADD("+"),
    AMP("&"),
    AND("AND"),
    DIV("/"),
    EQ("="),
    GT(">"),
    GTE(">="),
    LT("<"),
    LTE("<="),
    MOD("MOD"),
    MUL("*"),
    NEQ("<>"),
    OR("OR"),
    POW("**"),
    SUB("-"),
    XOR("XOR");

    private final String myAlias;

    BinaryOperation(String alias) {
        myAlias = alias;
    }

    public String getAlias() {
        return myAlias;
    }

    @Override
    public String toString() {
        return myAlias;
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/st/expressions/BinaryOperation.kt
package org.fbme.lib.st.expressions

enum class BinaryOperation(val alias: String, val luaAlias: String) {
    ADD("+", "+"), AMP("&", "&"), AND("AND", "and"), DIV("/", "/"),
    EQ("=", "=="), GT(">", ">"), GTE(">=", ">="), LT("<", "<"),
    LTE("<=", "<="), MOD("MOD", "%"), MUL("*", "*"), NEQ("<>", "~="),
    OR("OR", "or"), POW("**", "^"), SUB("-", "-"), XOR("XOR", "~");

    override fun toString(): String {
        return alias
    }
}
